src/internal.c

```diff
@@ -14928,31 +14937,29 @@ static word32 QSH_KeyExchangeWrite(WOLFSSL* ssl, byte isServer)
                         /* Create shared ECC key leaving room at the begining
                            of buffer for size of shared key. Note sizeof
                            preMasterSecret is ENCRYPT_LEN currently 512 */
-                        size = sizeof(ssl->arrays->preMasterSecret)
-                                                                 - OPAQUE16_LEN;
```

Comment:
Looks like "- OPAQUE16_LEN;" was dropped here. Need to adjust to make sure max size is correct.